openedx/adg/lms/applications/analytics.py

```diff
@@ -0,0 +1,191 @@
+            the end_date is `In Progress`.
+    """
+    start_date = get_formatted_date(
+        datetime(position.date_started_year, position.date_started_month, 1), DATE_FORMAT_FOR_MONTH_AND_YEAR_ONLY
+    )
+
+    is_position_current = (
+        (hasattr(position, 'is_current_position') and position.is_current_position) or
+        (hasattr(position, 'is_in_progress') and position.is_in_progress)
+    )
+    if is_position_current:
+        end_date = IN_PROGRESS
+    else:
+        end_date = get_formatted_date(
+            datetime(position.date_completed_year, position.date_completed_month, 1),
+            DATE_FORMAT_FOR_MONTH_AND_YEAR_ONLY
+        )
+
+    return start_date, end_date
+
+
+def emit_event_for_bu_courses_progress(user, status):
+    """
+    Emit event to update the progress of Business Unit courses
+
+    Arguments:
+        user (User): User for which to emit the BU courses progress update
+        status (str): Progress of the bu courses
+    """
+    segment.identify(user.id, {'businessLineCoursesStatus': status})
+
+
+def emit_course_progress_event(user, course, grade=None):
+    """
+    Emit the progress of a course when a course is Started, Completed or Failed. Also calls
+    `emit_enrollment_event_for_program_prereq_courses` function to emit an event if a program
+    prereq course is started or passed.
+
+    Arguments:
+        user (User): User associated with the course
+        course (CourseOverview): The course for which we are emitting the progress
+        grade (PersistentCourseGrade): The grade for the given course and user
+    """
+    from openedx.adg.lms.applications.helpers import has_attempted_all_modules
+
+    is_program_prereq_course = (
+        hasattr(course, 'multilingual_course') and
+        course.multilingual_course.multilingual_course_group.is_program_prerequisite
+    )
+
+    course_language_name = get_language_info(course.language).get('name', course.language)
+    course_properties = {'courseName': course.id.course, 'courseLanguage': course_language_name}
+
+    if grade:
+        course_properties['letterGrade'] = text_type(grade.letter_grade)
+        course_properties['percentageGrade'] = convert_float_point_to_percentage(grade.percent_grade)
+
+        if grade.passed_timestamp and grade.letter_grade:
+            course_event = COURSE_COMPLETED
+            if is_program_prereq_course:
+                emit_enrollment_event_for_program_prereq_courses(user.id, course, COURSE_COMPLETED)
+        else:
+            course_event = COURSE_FAILED if has_attempted_all_modules(user, course) else None
+    else:
+        course_event = COURSE_STARTED
+        if is_program_prereq_course:
+            emit_enrollment_event_for_program_prereq_courses(user.id, course, COURSE_STARTED)
+
+    if course_event:
+        segment.track(user.id, course_event, course_properties)
+
+
+def emit_enrollment_event_for_program_prereq_courses(user_id, course, course_status):
```

Comment:
This function emits on course enrolment as well as on course completion so let's please rename the function as right now it indicates that it only emits on enrollment.